### ISSUE TITLE
proposal: Add length assertions for FluentSlice, FluentMap, FluentString

### ISSUE THREAD

**Issue by esiqveland** [fluentassert/verify#123]
Hey, thanks for this great testing library! Really appreciate the slice asserts.

Would you consider adding som simpler assertions on slice length? 

Sometimes I just care that the resulting slice has length `x` or at least `x` elements.

Something like `verify.Slice(xs).AtLeastLen(3)...` `verify.Slice(xs).Len(3)....`

**Comment by pellared** [comment]
How about simply asseting the result of `len(xs)`

```
verify.Number(len(xs))...
```

?

**Comment by esiqveland** [comment]
This gives no opportunity for a good failure message that shows what was acrually in the list that was too short.

**Comment by pellared** [comment]
You can also use [a custom predicate](https://github.com/fluentassert/verify?tab=readme-ov-file#custom-predicates) e.g.

```go
verify.Slice(xs)..Should(func(x []X) bool {
		return len(xs) >= 3
	}).Assertf(t, "the slice is too short")
```
If you would need it more often then depending on your preference you can create [a custom assertion function](https://github.com/fluentassert/verify?tab=readme-ov-file#custom-assertion-function) or create your own "fluent assertion" as described in [here](https://github.com/fluentassert/verify?tab=readme-ov-file#extensibility).

If we would like to add "length assertions" for slices, then we should also extend the fluent assertions for `string` and `map` for the sake of consistency. I do not want to extend the API surface until there is a bigger demand.

**Comment by esiqveland** [comment]
Yeah, the `Should(...)` trick kind of works. You could think about adding a length specific helper! 👍

**Comment by pellared** [comment]
I started making a PR for `Lenght`, but I realized that I should also then make an "opposite" assertion. Therefore, we would need `LenghtEqual` and `LenghtNotEqual`. Then it would make sense to also add `LenghtGreater`, `LenghtGreaterOrEqual`, `LenghtLesser`, `LenghtLesserOrEqual`. But then it would make the API surface very (in my opinion unnecessary, big).

I think that simply adding `Len` (like [testify](https://pkg.go.dev/github.com/stretchr/testify/assert#Len)) would be good enough as it is the most critical assertion and the rest could be achieved using other techniques that I proposed in this issue.

@esiqveland, what do you think?

**Comment by esiqveland** [comment]
Yes, if you do decide to add it, I think starting with just `Len` makes most sense.

**Comment by pellared** [comment]
@esiqveland Released in v1.1.0
